core/src/main/java/org/elasticsearch/action/support/broadcast/node/TransportBroadcastByNodeAction.java

```diff
@@ -384,7 +384,11 @@ private void onShardOperation(final NodeRequest request, final Object[] shardRes
                 e.setIndex(shardRouting.getIndex());
                 e.setShard(shardRouting.shardId());
                 shardResults[shardIndex] = e;
-                logger.debug("[{}] failed to execute operation for shard [{}]", e, actionName, shardRouting.shortSummary());
+                if (logger.isDebugEnabled() && !TransportActions.isShardNotAvailableException(e)) {
```

Comment:
e != t :)

I think this will be simpler to read if it has a top level with TransportActions.isShardNotAvailableException(t) and then the right handling bellow it.